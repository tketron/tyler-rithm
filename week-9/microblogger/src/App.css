.App {
  display: flex;
  justify-content: space-around;
}

.App-posts {
  display: flex;
  flex-direction: column;
}

.App-NewPostForm {
  display: flex;
  flex-direction: column;
}
