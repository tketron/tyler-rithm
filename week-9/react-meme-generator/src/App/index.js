import React, { Component } from 'react';
import NewMemeForm from '../NewMemeForm';
import Meme from '../Meme';
import './style.css';

class App extends Component {
  state = {
    memes: []
  };

  addMeme = newMeme => {
    this.setState({
      memes: this.state.memes.concat(newMeme)
    });
  };

  deleteMeme = id => {
    this.setState(prevState => {
      return {
        memes: prevState.memes.filter(meme => meme.id !== id)
      };
    });
  };

  render() {
    return (
      <div className="App">
        <NewMemeForm addMeme={this.addMeme} />
        <hr />
        {this.state.memes.map(m => (
          <Meme key={m.id} {...m} deleteMeme={this.deleteMeme} />
        ))}
      </div>
    );
  }
}

export default App;
