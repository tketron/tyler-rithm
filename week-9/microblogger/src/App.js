import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import NewPostForm from './NewPostForm';
import PostList from './PostList';
import uuid from 'uuid/v4';
import TitleList from './TitleList';

class App extends Component {
  render() {
    return (
      <div className="App">
        <TitleList className="App-TitleList" />
        <div className="App-posts">
          <NewPostForm className="App-NewPostForm" />
          <PostList className="App-PostList" />
        </div>
      </div>
    );
  }
}

export default App;
