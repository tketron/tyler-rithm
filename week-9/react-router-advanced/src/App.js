import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import { Route, Switch, Redirect } from 'react-router-dom';

//mock authentication object
const fakeAuth = {
  isAuthenticated: false,
  authenticate(cb) {
    this.isAuthenticated = true;
    setTimeout(cb, 100); // fake async
  },
  signout(cb) {
    this.isAuthenticated = false;
    setTimeout(cb, 100); // fake async
  }
};

const Home = props => {
  return (
    <div>
      <h1>Homepage</h1>
      {/* <Route
        exact
        path="/taco"
        component={props => <small>nested routes work!</small>}
      /> */}
    </div>
  );
};

const About = props => {
  return <h1>About</h1>;
};

const Login = props => {
  return (
    <div>
      <h1>Login</h1>
    </div>
  );
};

const Protected = props => (
  <h1>This is a secret protected component that requires authentication.</h1>
);

const PrivateRoute = ({ component: Component, ...rest }) => {
  console.log(`Component: ${Component}, Rest: ${JSON.stringify(rest)}`);
  return (
    <Route
      {...rest}
      render={props =>
        fakeAuth.isAuthenticated ? (
          <Component {...props} />
        ) : (
          <Redirect to="/login" />
        )
      }
    />
  );
};

const NotFound = props => {
  return (
    <h1>{`${props.location.pathname} is not a valid route for this app`}</h1>
  );
};

class App extends Component {
  render() {
    return (
      <div className="App">
        <Switch>
          <Route exact path="/" component={Home} />
          <Route exact path="/about" component={About} />
          <Route exact path="/login" component={Login} />
          <PrivateRoute exact path="/protected" component={Protected} />
          <Route component={NotFound} />
          {/* <Redirect to="/" /> */}
        </Switch>
      </div>
    );
  }
}

export default App;
