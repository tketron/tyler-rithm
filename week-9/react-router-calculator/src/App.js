import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import { Route } from 'react-router-dom';

/**
 * Component to perform simple arithmetic calculations using query parameters
 * @param {*} props
 */
const Calculator = props => {
  // Destructure parameters
  const { operation, num1, num2 } = props.match.params;
  // Object containing basic arithmetic functions
  const calc = {
    add: (a, b) => a + b,
    subtract: (a, b) => a - b,
    divide: (a, b) => a / b,
    multiply: (a, b) => a * b
  };
  // Object containing the symbol version of the supported artihmetic operations
  const symbol = {
    add: '+',
    subtract: '-',
    divide: '/',
    multiply: '*'
  };
  // Test if parameter is a valid operation
  if (calc[operation]) {
    const result = calc[operation](+num1, +num2);
    return (
      <div>
        <h2>{`The result of ${num1} ${
          symbol[operation]
        } ${num2} is ${result}`}</h2>
      </div>
    );
  } else {
    return (
      <div>
        <h2>Not a valid operation!</h2>
      </div>
    );
  }
};

class App extends Component {
  render() {
    return (
      <div className="App">
        <Route path="/:operation/:num1/:num2" component={Calculator} />
      </div>
    );
  }
}

export default App;
