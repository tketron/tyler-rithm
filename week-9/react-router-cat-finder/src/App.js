import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import { Route, Switch, Redirect, Link } from 'react-router-dom';

const CatList = props => {
  console.log('cats list');
  return (
    <div>
      <Link to="/cats/cora">Cora</Link>
      <Link to="/cats/katie">Katie</Link>
      <Link to="/cats/carl">Carl</Link>
      <Route path="/cats/:name" component={Cat} />
    </div>
  );
};

const Cat = props => {
  console.log(JSON.stringify(props));
  const catFacts = {
    Cora: [
      'Cora loves attention',
      'Cora likes to kill butterflies',
      'Cora is actually a dog'
    ],
    Katie: [
      'Katie is a natural predator',
      'Katie is one of a kind',
      'Katie is reserved and dignified'
    ],
    Carl: [
      'Carl loves getting dirty',
      'Carl is a snuggle monster',
      'Carl has a very petite meow'
    ]
  };
  // let name = new URLSearchParams(props.match.params).get('name');
  let name = props.match.params.name;
  name = name.charAt(0).toUpperCase() + name.substr(1);
  return (
    <div>
      <p>The cat is {name}</p>
      <ul>{catFacts[name].map((f, idx) => <li key={idx}>{f}</li>)}</ul>
    </div>
  );
};

class App extends Component {
  render() {
    return (
      <div className="App">
        <Switch>
          <Route path="/cats" component={CatList} />
          <Redirect to="/cats" />
        </Switch>
      </div>
    );
  }
}

export default App;
