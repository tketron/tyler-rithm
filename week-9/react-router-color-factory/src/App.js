import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import { Route } from 'react-router-dom';

class App extends Component {
  render() {
    return (
      <div className="App">
        <Route path="/colors" component={Home} />
      </div>
    );
  }
}

export default App;
