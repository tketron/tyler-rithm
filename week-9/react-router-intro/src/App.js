import React, { Component } from 'react';
import { Route, Link } from 'react-router-dom';
import logo from './logo.svg';
import './App.css';

const Welcome = props => {
  return (
    <div>
      <h1>hello world!</h1>
    </div>
  );
};

const About = props => {
  return (
    <div>
      <h1>What about react router??</h1>
      <Link to="/">Go back home</Link>
    </div>
  );
};

const Contact = props => {
  return <h1>No contact, I'm a hermit.</h1>;
};

const Secret = props => {
  return (
    <div>
      <img src={props.gif} />
    </div>
  );
};

const Topics = props => {
  const stuff = ['politics', 'sports', 'music', 'coding'];
  return (
    <div>
      {stuff.map(t => (
        <div key={t}>
          <Link to={`/topics/${t}`}>{t}</Link>
        </div>
      ))}
      {stuff.map(t => (
        <Route
          key={t}
          exact
          path={`/topics/${t}`}
          component={props => <h1>{`The topic is ${t}`}</h1>}
        />
      ))}
    </div>
  );
};

class App extends Component {
  render() {
    return (
      <div>
        <ul>
          <li>
            <Link to="/">Home</Link>
          </li>
          <li>
            <Link to="/about">About</Link>
          </li>
          <li>
            <Link to="/contact">Contact</Link>
          </li>
          <li>
            <Link to="topics">Topics</Link>
          </li>
        </ul>
        <Route exact path="/" component={Welcome} />
        <Route path="/about" component={About} />
        <Route path="/contact" component={Contact} />
        <Route path="/topics" component={Topics} />
        {/* Passing down props through Routes */}
        <Route
          path="/secret"
          component={routeProps => (
            <Secret
              gif="https://media.giphy.com/media/kTBjwh6IWbLPy/giphy.gif"
              {...routeProps}
            />
          )}
        />
      </div>
    );
  }
}

export default App;
