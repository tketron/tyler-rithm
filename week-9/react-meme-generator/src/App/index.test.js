import React from 'react'; // for JSX
import { shallow } from 'enzyme'; // how to mount the component
import toJson from 'enzyme-to-json';
import App from '.'; // import the component itself

describe('<App />', () => {
  let wrapper;
  it('renders', () => {
    // smoke test!!!
    wrapper = shallow(<App />);
  });
  it('matches snapshot', () => {
    //snapshot test
    const serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
  });
});
