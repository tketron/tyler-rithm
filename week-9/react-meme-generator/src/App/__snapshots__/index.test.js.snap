// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> matches snapshot 1`] = `
<div
  className="App"
>
  <NewMemeForm
    addMeme={[Function]}
  />
  <hr />
</div>
`;
